packages/experiments/src/components/Pagination/Pagination.base.tsx

```diff
@@ -0,0 +1,150 @@
+            </button>
+          </li>
+        </ul>
+      </FocusZone>
+    );
+  }
+
+  private handleSelectedPage = (selected: number) => {
+    if (selected === this.props.selectedPageIndex) {
+      return; // same page, no action
+    }
+    if (this.props.onPageChange) {
+      this.props.onPageChange(selected);
+    }
+  };
+
+  private handlePreviousPage = () => {
+    this.handleSelectedPage(this.props.selectedPageIndex - 1);
+  };
+
+  private handleNextPage = () => {
+    this.handleSelectedPage(this.props.selectedPageIndex + 1);
+  };
+
+  private _pageElement(index: number, classNames: { [key in keyof IPaginationStyles]: string }): JSX.Element {
+    return (
+      <PageNumber
+        key={index + 1}
+        page={index + 1}
+        pageAriaLabel={this.props.pageAriaLabel}
+        selected={index === this.props.selectedPageIndex}
+        applyPage={this.handleSelectedPage}
+        className={index === this.props.selectedPageIndex ? classNames.selectedPageNumber : classNames.pageNumber}
+      />
+    );
+  }
+
+  private _pageList(classNames: { [key in keyof IPaginationStyles]: string }): JSX.Element[] {
+    const element = [];
+    if (this.props.pageCount <= this.props.pageRange!) {
+      for (let index = 0; index < this.props.pageCount; index++) {
+        element.push(this._pageElement(index, classNames));
+      }
+    } else {
+      let leftSide = this.props.pageRange! / 2;
+      let rightSide = this.props.pageRange! - leftSide;
+
+      if (this.props.selectedPageIndex > this.props.pageCount - 1 - this.props.pageRange! / 2) {
+        rightSide = this.props.pageCount - 1 - this.props.selectedPageIndex;
+        leftSide = this.props.pageRange! - rightSide;
+      } else if (this.props.selectedPageIndex < this.props.pageRange! / 2) {
+        leftSide = this.props.selectedPageIndex;
+        rightSide = this.props.pageRange! - leftSide;
+      }
+
+      let previousIndexIsOmitted = false;
+      for (let index = 0; index < this.props.pageCount; index++) {
+        const page = index + 1;
+        if (page <= this.props.marginPagesDisplayed!) {
+          element.push(this._pageElement(index, classNames));
+          previousIndexIsOmitted = false;
+          continue;
+        }
+
+        if (page > this.props.pageCount - this.props.marginPagesDisplayed!) {
+          element.push(this._pageElement(index, classNames));
+          previousIndexIsOmitted = false;
+          continue;
+        }
+
+        if (index >= this.props.selectedPageIndex - leftSide && index <= this.props.selectedPageIndex + rightSide) {
+          element.push(this._pageElement(index, classNames));
+          previousIndexIsOmitted = false;
+          continue;
+        }
+
+        if (previousIndexIsOmitted === false) {
+          const listKey = 'ellipsis' + index.toString();
+          element.push(
+            <li key={listKey} className="detailsListPagination-ellipsis" aria-label={this.props.omittedPagesAriaLabel}>
```

Comment:
We'll probably want this `className` generated via the `getStyles` approach so it can be targeted via styles API

Reply:
I removed it and will add another class for the omitlabel

---
In reply to: [222072962](https://github.com/OfficeDev/office-ui-fabric-react/pull/6532#discussion_r222072962) [](ancestors = 222072962)

Reply:
it is now removed. I will add another class to target the omit label

---
In reply to: [222072962](https://github.com/OfficeDev/office-ui-fabric-react/pull/6532#discussion_r222072962) [](ancestors = 222072962)